### ISSUE TITLE
feature: Include genre and album's date/year in listenbrainz scrobble payload

### ISSUE THREAD

**Issue by TomboFry** [sentriz/gonic#468]
Hello! I've been an avid user of Gonic for a few years now, and I'd really like for it to provide some additional fields when scrobbling to Listenbrainz. Specifically:

* Genre(s), via the `tags` field in `additional_info` (source: https://listenbrainz.readthedocs.io/en/latest/users/json.html#id1)
* Album's year/date, via the `date` field in `additional_info`.

I'm not 100% sure about including the second field, as I've not seen it referenced anywhere in the official ListenBrainz API docs, especially in the table linked above, nor could I really find much else online (_however_ in a very niche use-case it is something [a sample of a foobar extension](https://github.com/kbuffington/foo_jscript_panel/blob/8e8375c109595b18e624e8709ee6d84effba66b4/component/samples/complete/js/listenbrainz.js#L66) sends within its payload, which I've accidentally been fully embracing long before looking up the details to raise this issue). In any case, I'd be happy to exclude this one for the sake of fitting to the spec.

I believe these should be relatively easy implementations after having read [listenbrainz.go](https://github.com/sentriz/gonic/blob/314e9632d72dd7cd68044f9d69295123fff78f80/listenbrainz/listenbrainz.go#L47), and have absolutely considered making the contributions myself but am not at all used to the Go language and have struggled setting up the development environment to test it properly in the past 😅.

**Comment by sentriz** [comment]
hm I think this would need to be configurable. for example I use lastfm for tag genre metadata which quite often can be completely wrong (eg two artists with the same name)

I think it would be irresponsible for me for example to just start submitting those tags (some of which genres) to MB

where do you get your genre metadata as a matter of interest

**Comment by TomboFry** [comment]
> I use lastfm for tag genre metadata which quite often can be completely wrong

That's fair! I'm not sure how serious the field should be taken, considering the ListenBrainz documentation provides all metadata and MB UUIDs fields accurately for Never Gonna Give You Up, yet suggests `[ "you", "just", "got", "rick rolled!" ]` for the `tags` field 🤔 It also suggests the field is "A list of user-defined folksonomy tags to be associated with this recording"

> where do you get your genre metadata as a matter of interest ? 👀

Mostly manual - either from knowledge about the artist, and I'll use pretty consistent naming across my library, or from Wikipedia or MusicBrainz itself.

**Comment by sentriz** [comment]
yeah you can have nonsense tags like that fine, but if any of them happen to be also part of the list of predefined genres, then it comes a genre

 https://musicbrainz.org/tags

but yeah sure if people like yourself are manually adding genre info to your metatdat then this definitely makes sense. just the automatic sending of unverified potential genre tags to mb could hurt the data I reckon
